pkg/splunk/enterprise/indexercluster.go

```diff
@@ -194,6 +199,13 @@ func ApplyIndexerClusterManager(ctx context.Context, client splcommon.Controller
 		}
 	}
 
+	// check if the IndexerCluster is ready for version upgrade
+	cr.Kind = "IndexerCluster"
```

Comment:
Here as well?

Reply:
Mostly because when the mock test runs , its not setting `Kind` field in the structure. just to. make sure Unit test case works fine. but this is automatically set in run time